src/Cache/FileCache.php

```diff
@@ -190,6 +190,16 @@ public function onRequest(RequestEvent $requestEvent, string $eventName, EventDi
             $cacheFilePath = $this->getCacheFilePath($request, $nativeImage);
             $cacheFile = new File($cacheFilePath, false);
             $firstGen = false;
+
+            /**
+             * Also check date, if cached date is lower then original date -> Remove cached file
+             */
+            $mtime_original_file = $nativeImage->getRequestedFile()->getMTime();
```

Comment:
`getMTime` method can return `false` if file is invalid. This will occur when using `FlysystemFileResolver` with an external file system such as an *S3* Object storage.

You should test if `$mtime_original_file` is not `false` before comparing it to your cache file.

Reply:
https://www.php.net/manual/en/splfileinfo.getmtime.php

Reply:
NextGenFile should override `getMTime` to use `FilesystemReader::lastModified` method when `$filesystem` is not null

Reply:
https://flysystem.thephpleague.com/docs/usage/filesystem-api/